DEVELOPMENT.adoc

```diff
@@ -0,0 +1,180 @@
+Every change has to branch of from `main` and use this branch naming convention:
+
+* `feature/{type_of_change}-{short_description}` or with ticket id `feature/{ticket_id}/{type_of_change}-{short_description}`
+
+`main` must be always in releasable state.
+
+==== Type Of Change
+
+- *add* for new features or functionality
+- *change* for changes in existing features or functionality
+- *remove* | for removed features or functionality
+- *fix* for any bug fixes
+- *bump* for dependency updates
+- *security* in case of vulnerabilities
+
+Examples:
+
+- `feature/NAT-456/add-awesome-hashing-algorithm`
+- `feature/add-awesome-hashing-algorithm`
+- `feature/remove-not-so-awesome-algorithm`
+- `feature/fix-algorithm-corner-case`
+- `feature/bump-lib-to-1.3.0`
+
+== Release
+
+A release branches of from `main` branch with following pattern:
+
+* `release/{major}.{minor}(.{patch})/prepare-{major}.{minor}.{patch}` while patch is optional
+
+Add following changes:
+* [ ] Update CHANGELOG.adoc by creating a new Unreleased section and change current unreleased to release version
+* [ ] Update README.adoc `fhir-sdk-version` to release version
+
+== Develop
+
+The source is divided into supporting code for JSON parsing and establishing the FHIR primitive type system (String, Dates, Integer, etc.) for link:http://hl7.org/fhir/STU3/datatypes.html[FHIR 3]] and link:http://hl7.org/fhir/R4/datatypes.html[FHIR 4]]
+If you need to adjust one of the FHIR versions, it is located under `fhir/src/commonMain/kotlin` and the corresponding package.
+
+A big portion of the FHIR implementation is generated code, that could be found under `fhir/src-gen/commonMain/kotlin`. The generation is using link:https://github.com/gesundheitscloud/fhir-parser/[FHIR-Parser] and is controlled by `generateFhir.main.kts`.
+To trigger code generation just call `./generateFhir.main.kts` for all or add `fhir3` or `fhir4`.
+
+If you need to adjust the generated code, have a look at `fhir/parser/{FHIR version}/templates`:
+
+* `template-codesystems.kt`: the enums used to represent most FHIR type systems
+* `template-dependencies`: no used at the moment
+* `template-elementfactory.kt`: FhirHelper to define serialization and help type mapping
+* `template-resource`: FHIR model definition
+* `template-unittest`: Test to validate the FHIR implementation
+
+Additionally you could tweak the type mapping by altering `fhir/parser/{FHIR version}/config/mappings.py`.
+
+=== How to add a new FHIR version
+
+Add:
+. New FHIR spec under `fhir-spec/hl7.org/fhir/{new FHIR version}/`
+. New `config` and `templates` under `fhir/parser/{new FHIR version}/`
+.. Adjust them to your needs
+. Base implementation
+.. FHIR model base `Fhir{new FHIR version}`
+.. FHIR primitives
+.. `Fhir{new FHIR version}Parser` and register it in `FhirParserFactory` alongside with it's JSON implementation
+. new FHIR version handling to `generateFhir.main.kts`
+. generated models, tests and validation resources by triggering generation
```

Comment:
Also this should be bullet points, right? They are not shown correctly on GitHub.

Reply:
I'll change it to list style